### ISSUE TITLE
Support injecting already-connected workspace for pipeline build

### ISSUE THREAD

**Issue by jfomhover** [Azure/shrike#164]
**Category**
Please select:
- [x] shrike.pipeline
- [ ] shrike.build
- [ ] shrike.compliant_logging

Please also add corresponding labels.

The current implementation of AMLPipelineHelper.connect() and .run() makes it required to use shrike's aml arguments to provide references to the AzureML workspace.

https://github.com/Azure/shrike/blob/a588e07f99656c9681d5aaaff9eb6888b0143692/shrike/pipeline/pipeline_helper.py#L1327
https://github.com/Azure/shrike/blob/a588e07f99656c9681d5aaaff9eb6888b0143692/shrike/pipeline/pipeline_helper.py#L1398

It makes it impossible for developer to pre-connect to AzureML and provide an already connected workspace before building pipelines.

The reason this is important is because in some context you might want to build a pipeline within a process that already has its own workspace (from Run.get_context()).

**Describe the solution you'd like**
Two possible solutions:

1) modify the main() function to add an optional workspace object argument that would be passed to run(). The user would call main(workspace) to inject their own workspace.

2) to not use azureml_connect() in connect() if there's already a pre-existing workspace as returned by current_workspace(). The user would first call current_workspace(workspace) before any shrike call to set their workspace as the internal workspace the all pipeline builds.

**Describe alternatives you've considered**

**Estimate the Impact**
Any team using shrike within an AzureML experiment to build a pipeline from within an experiment. Not able to quantify this.

**Comment by dkmiller** [comment]
@jfomhover I ran into the same thing while writing the current orchestration logic for federated learning. If you want to see my current evil workaround, check out `azure_ml_connect` and `submittable_shrike_pipeline` in [business_logic.py](https://dev.azure.com/msdata/Vienna/_git/aml-ds?path=%2Frecipes%2Ffl_smart_reply%2Fbusiness_logic.py&version=GBmain&line=16&lineEnd=55&lineStartColumn=1&lineEndColumn=14&lineStyle=plain&_a=contents).

tl;dr &mdash; shrike is currently optimized for a single, _command line_, pipeline submission. We need to make it "play nicely" with multiple in-**Python** pipeline submissions, potentially to different workspaces.

Let's start with some design discussion :) .

**Comment by ShizeSu06** [comment]
Synced with Dan and Haozhe offline. I'm recently fully booked for some other high priority work items, but I expect to have some time/bandwidth to work on this item in late December after I come back from my vacation.
